rust-runtime/aws-smithy-runtime/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+pub mod config_bag;
+
+pub use crate::config_bag::ConfigBag;
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_runtime_api::interceptors::{InterceptorContext, Interceptors};
+use std::fmt::Debug;
+use std::future::Future;
+use std::pin::Pin;
+
+pub type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
+pub type BoxFallibleFut<T> = Pin<Box<dyn Future<Output = Result<T, BoxError>>>>;
+
+pub trait TraceProbe: Send + Sync + Debug {
```

Comment:
No need to change or anything and just sharing some thoughts; I am unsure about whether this trait (and other traits in this module) should live in `aws-smithy-runtime-api` or `aws-smithy-runtime` as it feels like they act as "interface"s. More broadly, though, I haven't been able to convince myself what goes into an interface crate and what goes into an implementation crate as that separation, _by design_, may not be as clear as in other languages, e.g., Java or C++.